src/ErrorHandler.php

```diff
@@ -397,19 +410,43 @@ private function handleFatalError(array $error = null): void
         }
 
         self::$reservedMemory = null;
+        $shouldExit = false;
 
         if (null === $error) {
             $error = error_get_last();
         }
 
+//        if (empty($error)) {
+//            return;
+//        }
+//
+//        if (
+//            self::$lastCapturedException
+//            && self::$lastCapturedException->getFile() === $error['file']
+//            && self::$lastCapturedException->getLine() === $error['line']
+//            && false !== strpos($error['message'], 'Uncaught ')
+//            && false !== strpos($error['message'], self::$lastCapturedException->getMessage())
+//        ) {
+//            return;
+//        }
+
         if (!empty($error) && $error['type'] & (E_ERROR | E_PARSE | E_CORE_ERROR | E_CORE_WARNING | E_COMPILE_ERROR | E_COMPILE_WARNING)) {
+            $shouldExit = true;
             $errorAsException = new FatalErrorException(self::ERROR_LEVELS_DESCRIPTION[$error['type']] . ': ' . $error['message'], 0, $error['type'], $error['file'], $error['line']);
 
             $this->exceptionReflection->setValue($errorAsException, []);
 
             $this->invokeListeners($this->errorListeners, $errorAsException);
             $this->invokeListeners($this->fatalErrorListeners, $errorAsException);
         }
+
+        if ($shouldExit) {
+            // Force the process to exit with code 255 to respect what would
+            // happen natively if no handler was registered
+            register_shutdown_function('register_shutdown_function', static function (): void {
```

Comment:
I don't like this, it's dangerous and it could kill the app before other shutdown functions are executed.

Why did you add this? The exit with failure was the only thing that worked even before.

Reply:
As the comment says, if we don't do this then in case of a fatal error the process will exit with code 0. Instead, natively, it would exit with code 255

Reply:
No it doesn't. The process was already exiting with 255, but without a fatal.